### PR TITLE
Update Deletions Of Events

### LINKED ISSUE #11: All events are not being deleted
After running del_events.py only current month's events are removed from calendar.

### LINKED ISSUE #17: del_events.py giving key error
```
if ("Class of " in event['summary'] or "Lab of " in event['summary']):
```

KeyError: 'summary'

### LINKED ISSUE #25: Del_events.py won't be able to select only gyft events
Previously, we searched for regex `Class of` or `Lab of` in `del_events.py` to choose only events added by `gyft` and then delete them. But after merging of PR #15 , this regex will no longer be valid. We need some other way to delete `gyft` events from Google Calendar.

Ping @icyflame @nishnik @DefCon-007 @kshitij10496

### LINKED ISSUE #67: Update the dates for 2020 Spring Semester
As is obvious by the name of the issue the repo needs to be updated with the dates of the 2020 spring semester.

The academic calendar for 19-20 can be found at http://www.iitkgp.ac.in/files/acad_cal1920.pdf

### PR DESCRIPTION
**Pull request by thealphadollar**
The PR brings the following changes which have been tested both using GYFT GUI and GYFT CLI add_events.py

- Add ability to delete events added via both ways: CLI or ICS.
- Freeze python requirements to avoid bugs.
- Update recurrences to use dates from dates.py and not have to manually enter them everytime updates are made.
- Add more visual output to avoid sense of no process.
- Show error message when wrong password entered.
- Update dates for the semester

closes #64 
closes #67 
fixes #25 
fixes #11 
fixes #17 
concerns #33

### CODE REVIEW

File: add_events.py [hunk 1]
```diff
@@ -120,7 +123,7 @@ def main():
             if (data[day][time][0] in subjects.keys()):
                 event['summary'] = subjects[data[day][time][0]].title()
             else: 
-                event['summary'] = data[day][time][0]
+                event['summary'] =  data[day][time][0]
```
Comment (icyflame) [inline add_events.py]: Not required

File: del_events.py [hunk 2]
```diff
@@ -44,28 +47,35 @@ def get_credentials():
         flow.user_agent = APPLICATION_NAME
         if flags:
             credentials = tools.run_flow(flow, store, flags)
-        else: # Needed only for compatibility with Python 2.6
+        else:  # Needed only for compatibility with Python 2.6
```
Comment (icyflame) [inline del_events.py]: Okay, I might have been mistaken in my previous comment. Is this some kind of python coding convention?

File: del_events.py [hunk 3]
```diff
@@ -44,28 +47,35 @@ def get_credentials():
...
+    # print(events)
     if not events:
         print('No upcoming events found.')
     for event in events:
-        if ("Class of " in event["summary"] or "Lab of " in event["summary"]):
-            service.events().delete(calendarId='primary', eventId=event["id"]).execute()
-            print ("Deleted: ", event["summary"], event["start"])
-    print ("Deletion done!")
+        # print(event.get('recurrence'))
+        if 'recurrence' in event and event['recurrence'] in [['RRULE:FREQ=WEEKLY;UNTIL={}'.format(END_TERM_BEGIN.strftime('%Y%m%dT000000Z'))],
```
Comment (icyflame) [inline del_events.py]: Store the repeated strftime format string in something like `constants.py` and import from there.

File: del_events.py [hunk 4]
```diff
@@ -44,28 +47,35 @@ def get_credentials():
...
+                                                            ['RRULE:FREQ=WEEKLY;UNTIL={}'.format(MID_TERM_BEGIN.strftime('%Y%m%dT000000Z'))],
+                                                            ['RRULE:FREQ=WEEKLY;UNTIL={}'.format(END_TERM_BEGIN.strftime('%Y%m%dT070000Z'))],
+                                                            ['RRULE:FREQ=WEEKLY;UNTIL={}'.format(MID_TERM_BEGIN.strftime('%Y%m%dT080000Z'))]]:
```
Comment (icyflame) [inline del_events.py]: This `if` clause is pretty long. Might be better to refactor into it's own method: `if is_gyft_event(event): service().delete(event)`

Comment (thealphadollar) [reply del_events.py]: Well, if we make it compact as mentioned in your previous review commit, I think it'll be manageable without a function.

File: add_events.py (hunk 1 above)
Comment (thealphadollar) [reply add_events.py]: Fixed.

File: del_events.py (hunk 2 above)
Comment (thealphadollar) [reply del_events.py]: Yes, it's a convention for comments.

File: del_events.py (hunk 3 above)
Comment (thealphadollar) [reply del_events.py]: Done.

**Pull request merged**